tests/manage/z_cluster/tier4_helpers.py

```diff
@@ -0,0 +1,202 @@
+import logging
+from concurrent.futures import ThreadPoolExecutor
+from uuid import uuid4
+
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import defaults
+from ocs_ci.ocs.ocp import OCP
+import ocs_ci.ocs.resources.pod as pod_helpers
+
+logger = logging.getLogger(__name__)
+
+
+def raw_block_io(raw_blk_pod,size='100G'):
+    """
+    Runs the block ios on pod baased raw block pvc
+    Args:
+        raw_blk_pod(pod): pod on which  block IOs should run
+        size(str): IO size
+
+    Returns:
+
+    """
+    raw_blk_pod.run_io(storage_type = 'block', size = size)
+    return True
+
+
+# move this function to cluster.py
+def get_osd_size():
+    """
+    Function to get the osd size in a cluster
+
+    Returns:
+        int: The size of the osd
+
+    """
+    ocp = OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE, kind=constants.STORAGECLUSTER)
+    sc = ocp.get()
+
+    # the following series of 'get's return capacity with size qualifier like Gi/Ti attached. So, take out last two
+    # characters to get actual capacity in int.
+    return sc.get('items')[0].get('spec').get('storageDeviceSets')[0].get('dataPVCTemplate').\
+        get('spec').get('resources').get('requests').get('storage')[:-2]
+
+
+def get_percent_used_capacity():
+
+    ct_pod = pod_helpers.get_ceph_tools_pod()
+    output = ct_pod.exec_ceph_cmd(ceph_cmd='ceph df')
+    total_used = (output.get('stats').get('total_used_raw_bytes'))
+    total_avail = (output.get('stats').get('total_bytes'))
+    return 100.0 * total_used / total_avail
+
+
+def downloader(dest_pod):
+    """
+     Downloads a wellknown file from the net on a given pod's /mnt directory
+
+    Args:
+        dest_pod: pod on which the file need to be downloaded
+
+    Returns:
+
+    """
+    curl_cmd = "" + "if [ -e /mnt/ceph.tar.gz ]; " \
+                    "then " \
+                    "   echo /mnt/ceph.tar.gz exists.; " \
+                    "else " \
+                    "   curl http://download.ceph.com/tarballs/ceph_15.1.0.orig.tar.gz --output /mnt/ceph.tar.gz ; " \
```

Comment:
please save the URL in a constants file.
Also, is there any other way of filling the cluster which will not require connecting to the remote URL?

Reply:
I had thought about it. we could copy from dir inside the pod root dir to /mnt/ dir. @shreekarSS please explore this. This would save time to download the file.